src/Http/Http.Extensions/gen/Microsoft.AspNetCore.Http.Generators.csproj

```diff
@@ -6,6 +6,7 @@
     <IsPackable>false</IsPackable>
     <IsAnalyzersProject>true</IsAnalyzersProject>
     <AddPublicApiAnalyzers>false</AddPublicApiAnalyzers>
+    <Nullable>enable</Nullable>
```

Comment:
Why isn't this the default in a Directory.Build.props?

Reply:
We haven't annotated the repo fully for nullability so our top-level targets only opt-in certain projects to it (see [this](https://github.com/dotnet/aspnetcore/blob/07937702255861896d9ec34e28c7c0c59ceb6a3f/eng/targets/CSharp.Common.targets#L16)). Setting the flag explicitly here imports the nullability attributes from netstandard projects.

Reply:
We have enabled it globally in the repo for "src" projects.
https://github.com/dotnet/aspnetcore/blob/07937702255861896d9ec34e28c7c0c59ceb6a3f/eng/targets/CSharp.Common.targets#L22
This project marks itself as an analyzer which means IsImplementationProject will be false.

Reply:
Can we just update that `CSharp.Common.targets` condition to include `IsGenProject`? (or whatever we are classifying this project as).

In dotnet/runtime we enable nullable for everything but test projects:

https://github.com/dotnet/runtime/blob/a84fff950a83c9c3a38e8da9eae89639f83b4f62/Directory.Build.props#L442-L448

Reply:
If you'd like to check if all current analyzer projects are nullable annotated, or make them nullable annotated sure :)

Reply:
> all current analyzer projects are nullable annotated

Why aren't they already? It's production code.

Reply:
It looks like this target isn't working the way that we expect either. The `<NoWarn>nullable</NoWarn>` is set as part of this target. Removing it reveals that we've got about ~150 nullability warnings in projects that we think have been annotated for nullability.

I filed https://github.com/dotnet/aspnetcore/issues/47179 for this and added a property to override this behavior in the build.

Reply:
Thanks for looking into this, @captainsafia.